realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -346,14 +310,58 @@ protected void onError(LogoutResponse response) {
     }
 
     /**
+     * Change this users password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
+     *
+     * @param newPassword the users new password.
+     * @throws IllegalArgumentException if not on a Looper thread.
```

Comment:
Ups, my bad (copy-paste). This does work on a non-looper thread, that is the entire purpose of this method.